gnmi_subscribe/gnmi_subscribe.go

```diff
@@ -176,11 +241,55 @@ func assembleSubscriptions(paths []*pb.Path) ([]*pb.Subscription, error) {
 		subscription := &pb.Subscription{
 			Path:              path,
 			Mode:              subscriptionMode,
-			SampleInterval:    *sampleInterval,
+			SampleInterval:    sampleInterval,
 			SuppressRedundant: *suppressRedundant,
 			HeartbeatInterval: *heartbeatInterval,
 		}
 		subscriptions = append(subscriptions, subscription)
 	}
 	return subscriptions, nil
 }
+
+func subscriptionMode(subscriptionPoll, subscriptionOnce bool) (gnmi.SubscriptionList_Mode, error) {
+	switch {
+	case subscriptionPoll && subscriptionOnce:
+		return 0, errors.New("Only one of -once and -poll can be set")
+	case subscriptionOnce:
+		return pb.SubscriptionList_ONCE, nil
+	case subscriptionPoll:
+		return pb.SubscriptionList_POLL, nil
+	default:
+		return pb.SubscriptionList_STREAM, nil
+	}
+}
+
+func parsePaths(xPathFlags, pbPathFlags arrayFlags) ([]*pb.Path, error) {
+	var pbPathList []*pb.Path
+	for _, xPath := range xPathFlags {
+		pbPath, err := xpath.ToGNMIPath(xPath)
+		if err != nil {
+			return nil, fmt.Errorf("error in parsing xpath %q to gnmi path", xPath)
+		}
+		pbPathList = append(pbPathList, pbPath)
+	}
+	for _, textPbPath := range pbPathFlags {
+		var pbPath pb.Path
+		if err := proto.UnmarshalText(textPbPath, &pbPath); err != nil {
+			return nil, fmt.Errorf("error in unmarshaling %q to gnmi Path", textPbPath)
+		}
+		pbPathList = append(pbPathList, &pbPath)
+	}
+	return pbPathList, nil
+}
+
+func parseEncoding(encodingFormat string) (gnmi.Encoding, error) {
+	encoding, ok := pb.Encoding_value[encodingFormat]
+	if !ok {
+		var encodingList []string
+		for _, name := range pb.Encoding_name {
+			encodingList = append(encodingList, name)
+		}
+		return -1, fmt.Errorf("Supported encodings: %s", strings.Join(encodingList, ", "))
```

Comment:
Similar to earlier,  could we create a new Encoding yupe of UNKNOWN here:

https://github.com/openconfig/gnmi/blob/964044615d9f0a475afb220fdae5026be1a75ee6/proto/gnmi/gnmi.proto#L172